baremaps-openapi/src/main/java/com/baremaps/openapi/TilesetQueryParser.java

```diff
@@ -0,0 +1,273 @@
+package com.baremaps.openapi;
+
+import com.baremaps.model.Layer;
+import com.baremaps.model.Query;
+import com.baremaps.model.TileSet;
+import com.baremaps.tile.Tile;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import net.sf.jsqlparser.JSQLParserException;
+import net.sf.jsqlparser.expression.Expression;
+import net.sf.jsqlparser.expression.Parenthesis;
+import net.sf.jsqlparser.expression.operators.conditional.OrExpression;
+import net.sf.jsqlparser.parser.CCJSqlParserUtil;
+import net.sf.jsqlparser.statement.select.FromItem;
+import net.sf.jsqlparser.statement.select.Join;
+import net.sf.jsqlparser.statement.select.PlainSelect;
+import net.sf.jsqlparser.statement.select.Select;
+import net.sf.jsqlparser.statement.select.SelectExpressionItem;
+import net.sf.jsqlparser.statement.select.SelectItem;
+import net.sf.jsqlparser.statement.select.SelectItemVisitorAdapter;
+
+public class TilesetQueryParser {
```

Comment:
This duplicated code will be removed once we have a clear separation between domain and transfer objects (#316).